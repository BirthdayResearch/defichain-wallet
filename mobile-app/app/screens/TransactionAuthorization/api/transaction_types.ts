export const MAX_PASSCODE_ATTEMPT = 3 // allowed 2 failures
export const PASSCODE_LENGTH = 6
export const INVALID_HASH = 'invalid hash'
export const USER_CANCELED = 'USER_CANCELED'

export const DEFAULT_MESSAGES = {
  message: 'Enter passcode to continue',
  loadingMessage: 'Signing your transaction...',
  authorizedTransactionMessage: {
    title: 'Transaction authorized',
    description: 'Please wait while your transaction is being prepared'
  },
  grantedAccessMessage: {
    title: 'Access granted',
    description: 'You may now proceed'
  }
}

export const SUCCESS_DISPLAY_TIMEOUT_IN_MS = 2000
export const CANCELED_ERROR = 'canceled error'

export enum TransactionStatus {
  INIT,
  IDLE,
  BLOCK,
  PIN,
  SIGNING,
  AUTHORIZED
}

export interface PromptPromiseI {
  resolve: (pin: string) => void
  reject: (e: Error) => void
}
