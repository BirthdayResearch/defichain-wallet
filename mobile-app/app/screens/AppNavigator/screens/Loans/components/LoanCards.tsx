
import { ThemedFlatList, ThemedIcon, ThemedText, ThemedTouchableOpacity } from '@components/themed'
import { tailwind } from '@tailwind'
import { translate } from '@translations'
import NumberFormat from 'react-number-format'
import { View } from 'react-native'
import { getNativeIcon } from '@components/icons/assets'
import { LoanToken, LoanVaultActive, LoanVaultState } from '@defichain/whale-api-client/dist/api/loan'
import { NavigationProp, useNavigation } from '@react-navigation/native'
import { LoanParamList } from '../LoansNavigator'
import { ActivePrice } from '@defichain/whale-api-client/dist/api/prices'
import { useSelector } from 'react-redux'
import { RootState } from '@store'
import { vaultsSelector } from '@store/loans'
import { getUSDPrecisedPrice } from '@screens/AppNavigator/screens/Auctions/helpers/usd-precision'
import { getActivePrice } from '../../Auctions/helpers/ActivePrice'

interface LoanCardsProps {
  loans: LoanToken[]
  testID?: string
  vaultId?: string
}

export interface LoanCardOptions {
  loanTokenId: string
  symbol: string
  displaySymbol: string
  price?: ActivePrice
  interestRate: string
  onPress: () => void
  testID: string
}

export function LoanCards (props: LoanCardsProps): JSX.Element {
  const navigation = useNavigation<NavigationProp<LoanParamList>>()
  const vaults = useSelector((state: RootState) => vaultsSelector(state.loans))
  const activeVault = vaults.find((v) => v.vaultId === props.vaultId && v.state !== LoanVaultState.IN_LIQUIDATION) as LoanVaultActive
  return (
    <>
      <ThemedFlatList
        contentContainerStyle={tailwind('px-2 pt-4 pb-2')}
        data={props.loans}
        numColumns={2}
        renderItem={({
          item,
          index
        }: { item: LoanToken, index: number }): JSX.Element => {
          return (
            <View style={{ flexBasis: '50%' }}>
              <LoanCard
                symbol={item.token.symbol}
                displaySymbol={item.token.displaySymbol}
                interestRate={item.interest}
                price={item.activePrice}
                loanTokenId={item.tokenId}
                onPress={() => {
                  navigation.navigate({
                    name: 'BorrowLoanTokenScreen',
                    params: {
                      loanToken: item,
                      vault: activeVault
                    },
                    merge: true
                  })
                }}
                testID={`loan_card_${index}`}
              />
            </View>
          )
        }}
        keyExtractor={(_item, index) => index.toString()}
        testID={props.testID}
      />
    </>
  )
}

function LoanCard ({
  symbol,
  displaySymbol,
  price,
  interestRate,
  onPress,
  testID
}: LoanCardOptions): JSX.Element {
  const LoanIcon = getNativeIcon(displaySymbol)
  const currentPrice = getUSDPrecisedPrice(getActivePrice(symbol, price))
  return (
    <ThemedTouchableOpacity
      testID={`loan_card_${displaySymbol}`}
      light={tailwind('bg-white border-gray-200')}
      dark={tailwind('bg-gray-800 border-gray-700')}
      style={tailwind('p-4 mx-2 mb-4 rounded flex-1 border')}
      onPress={onPress}
    >
      <View style={tailwind('flex-row items-center pb-2 justify-between')}>
        <View style={tailwind('flex flex-row')}>
          <LoanIcon width={24} height={24} style={tailwind('mr-2')} />
          <ThemedText testID={`${testID}_display_symbol`} style={tailwind('font-medium')}>{displaySymbol}</ThemedText>
        </View>
        <ThemedIcon iconType='MaterialIcons' name='chevron-right' size={20} style={tailwind('-mr-2')} />
      </View>
      <ThemedText
        light={tailwind('text-gray-500')}
        dark={tailwind('text-gray-400')}
        style={tailwind('text-xs')}
      >
        {translate('components/LoanCard', 'Interest')}
      </ThemedText>
      <NumberFormat
        decimalScale={2}
        thousandSeparator
        displayType='text'
        renderText={(value) =>
          <ThemedText testID={`${testID}_interest_rate`} style={tailwind('text-sm pb-2')}>
            {value}
          </ThemedText>}
        value={interestRate}
        suffix='%'
      />
      <ThemedText
        light={tailwind('text-gray-500')}
        dark={tailwind('text-gray-400')}
        style={tailwind('text-xs')}
      >
        {translate('components/LoanCard', 'Price (USD)')}
      </ThemedText>
      <NumberFormat
        decimalScale={2}
        thousandSeparator
        displayType='text'
        renderText={(value) =>
          <View style={tailwind('flex flex-row items-center')}>
            <ThemedText testID={`${testID}_loan_amount`} style={tailwind('text-sm mr-1')}>
              ${value}
            </ThemedText>
          </View>}
        value={currentPrice}
      />
    </ThemedTouchableOpacity>
  )
}
