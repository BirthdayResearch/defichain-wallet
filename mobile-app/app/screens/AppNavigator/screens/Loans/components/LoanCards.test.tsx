import { render } from '@testing-library/react-native'
import { LoanCards } from './LoanCards'
import { LoanToken, LoanVaultState } from '@defichain/whale-api-client/dist/api/loan'
import { Provider } from 'react-redux'
import { RootState } from '@store'
import { configureStore } from '@reduxjs/toolkit'
import { loans } from '@store/loans'
import { wallet } from '@store/wallet'

jest.mock('@shared-contexts/ThemeProvider')
jest.mock('@contexts/FeatureFlagContext')
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn()
}))

describe('loan cards', () => {
  it('should match snapshot', async () => {
    const TS25 = {
      id: '17',
      symbol: 'TS25',
      symbolKey: 'TS25',
      name: '',
      decimal: 8,
      limit: '0',
      mintable: true,
      tradeable: true,
      isDAT: true,
      isLPS: false,
      isLoanToken: true,
      finalized: false,
      minted: '0',
      creation: {
        tx: '3a7e97db4b913fd249da2a59f2edd84f34e111fe1c775a01addfb3b96c147d40',
        height: 151
      },
      destruction: {
        tx: '0000000000000000000000000000000000000000000000000000000000000000',
        height: -1
      },
      collateralAddress: 'bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny',
      displaySymbol: 'dTS25'
    }

    const TR50 = {
      id: '18',
      symbol: 'TR50',
      symbolKey: 'TR50',
      name: '',
      decimal: 8,
      limit: '0',
      mintable: true,
      tradeable: true,
      isDAT: true,
      isLPS: false,
      isLoanToken: true,
      finalized: false,
      minted: '0',
      creation: {
        tx: '57135919430b915f6f462142161e94474aae824791fa3644541102d7a7959a63',
        height: 152
      },
      destruction: {
        tx: '0000000000000000000000000000000000000000000000000000000000000000',
        height: -1
      },
      collateralAddress: 'bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny',
      displaySymbol: 'dTR50'
    }

    const TU10 = {
      id: '15',
      symbol: 'TU10',
      symbolKey: 'TU10',
      name: '',
      decimal: 8,
      limit: '0',
      mintable: true,
      tradeable: true,
      isDAT: true,
      isLPS: false,
      isLoanToken: true,
      finalized: false,
      minted: '0',
      creation: {
        tx: '943c2f38cf418ff116c402f4f6ec400039223e5aa4fb41f2c4a6e69ac53a4c45',
        height: 149
      },
      destruction: {
        tx: '0000000000000000000000000000000000000000000000000000000000000000',
        height: -1
      },
      collateralAddress: 'bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny',
      displaySymbol: 'dTU10'
    }

    const loanCards: LoanToken[] = [
      {
        tokenId: '3a7e97db4b913fd249da2a59f2edd84f34e111fe1c775a01addfb3b96c147d40',
        token: TS25,
        interest: '2',
        fixedIntervalPriceId: 'TS25/USD'
      },
      {
        tokenId: '57135919430b915f6f462142161e94474aae824791fa3644541102d7a7959a63',
        token: TR50,
        interest: '3',
        fixedIntervalPriceId: 'TR50/USD'
      },
      {
        tokenId: '943c2f38cf418ff116c402f4f6ec400039223e5aa4fb41f2c4a6e69ac53a4c45',
        token: TU10,
        interest: '1',
        fixedIntervalPriceId: 'TU10/USD'
      }
    ]
    const initialState: Partial<RootState> = {
      loans: {
        vaults: [{
          vaultId: '22ffasd5ca123123123123123121231061',
          loanAmounts: [],
          collateralRatio: '',
          collateralValue: '',
          collateralAmounts: [],
          loanScheme: {
            id: '0',
            interestRate: '3',
            minColRatio: '100'
          },
          loanValue: '100',
          ownerAddress: 'bcrt1qxzj8pnkeqznvx6xgeepdywus8lkxq3vvmeccyt',
          state: LoanVaultState.ACTIVE,
          informativeRatio: '0',
          interestAmounts: [],
          interestValue: '1'
        }],
        collateralTokens: [],
        hasFetchedLoansData: false,
        hasFetchedVaultsData: false,
        loanSchemes: [],
        loanTokens: []
      },
      wallet: {
        utxoBalance: '77',
        tokens: [],
        allTokens: { dTS25: TS25, dTR50: TR50, dTU10: TU10 },
        poolpairs: [],
        hasFetchedPoolpairData: false
      }
    }
    const store = configureStore({
      preloadedState: initialState,
      reducer: {
        wallet: wallet.reducer,
        loans: loans.reducer
      }
    })
    const rendered = render((<Provider store={store}><LoanCards loans={loanCards} /></Provider>))
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
