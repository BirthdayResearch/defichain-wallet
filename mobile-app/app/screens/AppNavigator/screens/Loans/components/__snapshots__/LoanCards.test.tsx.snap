// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loan cards should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 16,
    }
  }
  data={
    Array [
      Object {
        "fixedIntervalPriceId": "TS25/USD",
        "interest": "2",
        "token": Object {
          "collateralAddress": "bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny",
          "creation": Object {
            "height": 151,
            "tx": "3a7e97db4b913fd249da2a59f2edd84f34e111fe1c775a01addfb3b96c147d40",
          },
          "decimal": 8,
          "destruction": Object {
            "height": -1,
            "tx": "0000000000000000000000000000000000000000000000000000000000000000",
          },
          "displaySymbol": "dTS25",
          "finalized": false,
          "id": "17",
          "isDAT": true,
          "isLPS": false,
          "isLoanToken": true,
          "limit": "0",
          "mintable": true,
          "minted": "0",
          "name": "",
          "symbol": "TS25",
          "symbolKey": "TS25",
          "tradeable": true,
        },
        "tokenId": "3a7e97db4b913fd249da2a59f2edd84f34e111fe1c775a01addfb3b96c147d40",
      },
      Object {
        "fixedIntervalPriceId": "TR50/USD",
        "interest": "3",
        "token": Object {
          "collateralAddress": "bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny",
          "creation": Object {
            "height": 152,
            "tx": "57135919430b915f6f462142161e94474aae824791fa3644541102d7a7959a63",
          },
          "decimal": 8,
          "destruction": Object {
            "height": -1,
            "tx": "0000000000000000000000000000000000000000000000000000000000000000",
          },
          "displaySymbol": "dTR50",
          "finalized": false,
          "id": "18",
          "isDAT": true,
          "isLPS": false,
          "isLoanToken": true,
          "limit": "0",
          "mintable": true,
          "minted": "0",
          "name": "",
          "symbol": "TR50",
          "symbolKey": "TR50",
          "tradeable": true,
        },
        "tokenId": "57135919430b915f6f462142161e94474aae824791fa3644541102d7a7959a63",
      },
      Object {
        "fixedIntervalPriceId": "TU10/USD",
        "interest": "1",
        "token": Object {
          "collateralAddress": "bcrt1qyrfrpadwgw7p5eh3e9h3jmu4kwlz4prx73cqny",
          "creation": Object {
            "height": 149,
            "tx": "943c2f38cf418ff116c402f4f6ec400039223e5aa4fb41f2c4a6e69ac53a4c45",
          },
          "decimal": 8,
          "destruction": Object {
            "height": -1,
            "tx": "0000000000000000000000000000000000000000000000000000000000000000",
          },
          "displaySymbol": "dTU10",
          "finalized": false,
          "id": "15",
          "isDAT": true,
          "isLPS": false,
          "isLoanToken": true,
          "limit": "0",
          "mintable": true,
          "minted": "0",
          "name": "",
          "symbol": "TU10",
          "symbolKey": "TU10",
          "tradeable": true,
        },
        "tokenId": "943c2f38cf418ff116c402f4f6ec400039223e5aa4fb41f2c4a6e69ac53a4c45",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexBasis": "50%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderRightWidth": 1,
                "borderTopColor": "rgba(229, 229, 229, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
                "borderTopWidth": 1,
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 16,
                "marginLeft": 8,
                "marginRight": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="loan_card_dTS25"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={32}
                  vbWidth={32}
                  width={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      clipPath="clip0_1388_11287"
                    >
                      <RNSVGPath
                        clipRule={0}
                        d="M16 0C7.163 0 0 7.163 0 16C0 24.837 7.163 32 16 32C24.838 32 32 24.837 32 16C32 7.163 24.838 0 16 0Z"
                        fill={4279110157}
                        fillRule={0}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      />
                    </RNSVGGroup>
                    <RNSVGText
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 10,
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "62%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="TS25"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Object {
                          "fontFamily": "MediumFont",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 1)",
                        },
                      ],
                    ]
                  }
                  testID="loan_card_0_display_symbol"
                >
                  dTS25
                </Text>
              </View>
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Interest
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="loan_card_0_interest_rate"
            >
              2%
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Price (USD)
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "lineHeight": 20,
                        "marginRight": 4,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
                testID="loan_card_0_loan_amount"
              >
                $
                0.00
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexBasis": "50%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderRightWidth": 1,
                "borderTopColor": "rgba(229, 229, 229, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
                "borderTopWidth": 1,
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 16,
                "marginLeft": 8,
                "marginRight": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="loan_card_dTR50"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={32}
                  vbWidth={32}
                  width={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      clipPath="clip0_1388_11287"
                    >
                      <RNSVGPath
                        clipRule={0}
                        d="M16 0C7.163 0 0 7.163 0 16C0 24.837 7.163 32 16 32C24.838 32 32 24.837 32 16C32 7.163 24.838 0 16 0Z"
                        fill={4279110157}
                        fillRule={0}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      />
                    </RNSVGGroup>
                    <RNSVGText
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 10,
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "62%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="TR50"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Object {
                          "fontFamily": "MediumFont",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 1)",
                        },
                      ],
                    ]
                  }
                  testID="loan_card_1_display_symbol"
                >
                  dTR50
                </Text>
              </View>
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Interest
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="loan_card_1_interest_rate"
            >
              3%
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Price (USD)
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "lineHeight": 20,
                        "marginRight": 4,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
                testID="loan_card_1_loan_amount"
              >
                $
                0.00
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexBasis": "50%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderLeftWidth": 1,
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderRightWidth": 1,
                "borderTopColor": "rgba(229, 229, 229, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
                "borderTopWidth": 1,
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 16,
                "marginLeft": 8,
                "marginRight": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="loan_card_dTU10"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={32}
                  vbWidth={32}
                  width={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      clipPath="clip0_1388_11287"
                    >
                      <RNSVGPath
                        clipRule={0}
                        d="M16 0C7.163 0 0 7.163 0 16C0 24.837 7.163 32 16 32C24.838 32 32 24.837 32 16C32 7.163 24.838 0 16 0Z"
                        fill={4279110157}
                        fillRule={0}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      />
                    </RNSVGGroup>
                    <RNSVGText
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 10,
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "62%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="TU10"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Object {
                          "fontFamily": "MediumFont",
                          "fontWeight": "500",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 1)",
                        },
                      ],
                    ]
                  }
                  testID="loan_card_2_display_symbol"
                >
                  dTU10
                </Text>
              </View>
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Interest
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
              testID="loan_card_2_interest_rate"
            >
              1%
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Price (USD)
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "lineHeight": 20,
                        "marginRight": 4,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
                testID="loan_card_2_loan_amount"
              >
                $
                0.00
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
